tests/sklearn/test_sklearn_model_export.py

```diff
@@ -874,3 +876,47 @@ def test_model_registration_metadata_handling(sklearn_knn_model, tmp_path):
     # This validates that the models artifact repo will not attempt to create a
     # "registered model metadata" file if the source of an artifact download is a file.
     assert os.listdir(dst_full) == ["MLmodel"]
+
+
+@pytest.mark.skipif(
+    Version(sklearn.__version__) < Version("1.0.0"),
+    reason="predict_proba method patching for test doesn't work in scikit-learn < 1",
+)
+def test_pipeline_model_wrapper_prediction(sklearn_knn_model, model_path):
+    # Test `_SklearnModelWrapper` supports sklearn pipeline model with arbitrary params.
+    # Related issue: https://github.com/mlflow/mlflow/issues/12445
+    knn_model = sklearn_knn_model.model
+
+    predict_invocation_params = []
+    original_predict = knn_model.predict
+
+    def predict(data, p1=None):
+        predict_invocation_params.append(p1)
+        return original_predict(data)
+
+    knn_model.predict = predict
```

Comment:
why do we need do this?

Reply:
can we use the model in #12445?

Reply:
We can . but the model in https://github.com/mlflow/mlflow/issues/12445 doesn't have additional inference params to test (i.e. in this case the passed in `params` argument is None)

Reply:
let's use a more realistic example.